storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/BlindedBlockMigration.java

```diff
@@ -107,4 +129,167 @@ private void moveFirstFinalizedBlockToBlindedStorage() {
           }
         });
   }
+
+  private void migrateRemainingBlocks() {
+    LOG.debug("migrate finalized un-blinded blocks to blinded storage.");
+    final long finalizedBlockIndexCounter = copyFinalizedBlockIndexToBlindedStorage();
+    final long preBellatrixMigratedBlocks =
+        migrateFinalizedBlocksPreBellatrix(finalizedBlockIndexCounter);
+    migrateFinalizedBlocks(finalizedBlockIndexCounter - preBellatrixMigratedBlocks);
+    migrateNonCanonicalBlocks();
+  }
+
+  private long copyFinalizedBlockIndexToBlindedStorage() {
+    long counter = 0;
+    try (final Stream<Map.Entry<Bytes32, UInt64>> blockRoots =
+        dao.streamUnblindedFinalizedBlockRoots()) {
+      for (Iterator<Map.Entry<Bytes32, UInt64>> it = blockRoots.iterator(); it.hasNext(); ) {
+        if (counter == 0) {
+          LOG.info("Copying finalized block indices to blinded storage");
+        }
+        try (KvStoreCombinedDaoBlinded.FinalizedUpdaterBlinded finalizedUpdaterBlinded =
+            dao.finalizedUpdaterBlinded()) {
+          for (int i = 0; i < INDEX_BATCH_SIZE && it.hasNext(); i++) {
+            final Map.Entry<Bytes32, UInt64> entry = it.next();
+            final Bytes32 root = entry.getKey();
+            final UInt64 slot = entry.getValue();
+            finalizedUpdaterBlinded.addFinalizedBlockRootBySlot(slot, root);
+            counter++;
+          }
+          finalizedUpdaterBlinded.commit();
+        }
+
+        pause();
+      }
+    }
+    if (counter > 0) {
+      LOG.info("{} finalized block indices copied", counter);
+    }
+    return counter;
+  }
+
+  private long migrateFinalizedBlocksPreBellatrix(final long migrationCounter) {
+    long counter = 0;
+    boolean preBellatrix = true;
+    if (migrationCounter > 0) {
+      LOG.info(
+          "Migrating pre-bellatrix blocks to blinded storage, {} finalized blocks to migrate",
+          migrationCounter);
```

Comment:
the `XX finalized blocks to migrate` is all finalized blocks, so some may not be copied as part of this step, not sure if that's misleading...
The other finalized migrate will say `XX blocks to migrate`, which is based on the total number of finalized blocks in the index minus the number migrated by this process..